src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala

```diff
@@ -591,14 +593,14 @@ class GroupReadsByUmi
   /** Consumes the next group of templates with all matching end positions and returns them. */
   def takeNextGroup(iterator: BufferedIterator[Template]) : Seq[Template] = {
     val first     = iterator.next()
-    val firstEnds = ReadInfo(first.r1.getOrElse(fail(s"R1 missing for template ${first.name}")))
+    val firstEnds = ReadInfo(first.r1.getOrElse(fail(s"R1 missing for template ${first.name}")), ignoreStrand=first.unpaired)
     val firstUmi  = first.r1.get.apply[String](this.assignTag)
     val builder   = IndexedSeq.newBuilder[Template]
     builder    += first
 
     while (
       iterator.hasNext &&
-      firstEnds == ReadInfo(iterator.head.r1.get) &&
+      firstEnds == ReadInfo(iterator.head.r1.get, ignoreStrand=iterator.head.unpaired) &&
```

Comment:
This is also unsafe ... for performance reasons `ReadInfo` stores the value in a transient attribute on the record.  If you call it twice with different values for `ignoreStrand` you'll get the wrong answer the second time.